pkg/auth/authproviders/oidc/backend_impl.go

```diff
@@ -629,9 +653,69 @@ func userInfoToExternalClaims(userInfo *userInfoType) *tokens.ExternalUserClaim
 	return claim
 }
 
-func (p *backendImpl) injectHTTPClient(ctx context.Context) context.Context {
-	if p.httpClient != nil {
-		return oidc.ClientContext(ctx, p.httpClient)
+type claimExtractor interface {
+	Claims(v interface{}) error
+}
+
+func extractCustomClaims(externalUserClaim *tokens.ExternalUserClaim, mappings map[string]string, claimExtractor claimExtractor) error {
+	claims := make(map[string]interface{}, 0)
+	if err := claimExtractor.Claims(&claims); err != nil {
+		return errors.Wrap(err, "failed to construct Go map")
 	}
-	return ctx
+	for fromClaimName, toClaimName := range mappings {
+		val, err := extractClaimFromPath(fromClaimName, claims)
+		if err != nil {
+			return err
+		}
+		if err := addClaimToUserClaims(externalUserClaim, toClaimName, val); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func addClaimToUserClaims(externalUserClaim *tokens.ExternalUserClaim, attributeName string, claimValue interface{}) error {
+	switch v := claimValue.(type) {
+	case []interface{}:
+		for i, arrayVal := range v {
+			_, isArray := arrayVal.([]interface{})
```

Comment:
In the PR description stays that we support nested arrays, but I didn't see that functionality. By nested arrays, I would assume the mapping of:

with path: `arr`
```
{
  "arr": [
    ["one", "two"]
  ]
}
```

or with path `arr.num`:
```
{
  "arr": [
    { "num": "one" }
  ]
}
```

What is meant by nested array? Or have I missed something in the code? 😊

Reply:
To be honest I didn't figure out right vocabulary yet, so it can be confusing what is possible and what isn't. I'll add examples to the PR description.